include/dlaf/eigensolver/reduction_to_band/mc.h

```diff
@@ -308,6 +321,223 @@ void trmmComputeW(PanelT<Coord::Col, T>& w, MatrixLikeT& v, hpx::shared_future<C
+      const LocalTileIndex ij{i, j};
+
+      const bool is_diagonal_tile = (ij.row() == ij.col());
+
+      if (is_diagonal_tile) {
+        hpx::dataflow(ex, unwrapExtendTiles(hemm_o), blas::Side::Left, blas::Uplo::Lower, T(1),
+                      a.read(ij), w.read(ij), T(1), x(ij));
+      }
+      else {
+        // A . W*
+        {
+          // Note:
+          // Since it is not a diagonal tile, otherwise it would have been managed in the previous
+          // branch, the second operand is not available in W but it is accessible through the
+          // support panel Wt.
+          // However, since we are still computing the "straight" part, the result can be stored
+          // in the "local" panel X.
+
+          hpx::dataflow(ex, unwrapExtendTiles(gemm_o), blas::Op::NoTrans, blas::Op::NoTrans, T(1),
+                        a.read(ij), w.read(transposed(ij)), T(1), x(ij));
+        }
+
+        // A* . W
+        {
+          // Note:
+          // Here we are considering the hermitian part of A, so coordinates have to be "mirrored".
+          // So, first step is identifying the mirrored cell coordinate, i.e. swap row/col, together
+          // with realizing if the new coord lays on an owned row or not.
+          // If yes, the result can be stored in the X, otherwise Xt support panel will be used.
+          // For what concerns the second operand, it can be found for sure in W. In fact, the
+          // multiplication requires matching col(A) == row(W), but since coordinates are mirrored,
+          // we are mathing row(A) == row(W), so it is local by construction.
+          const auto owner = dist.template rankGlobalTile<Coord::Row>(ij.col());
+
+          // X is in the lower part but it is ConjTrans
+          hpx::dataflow(ex, unwrapExtendTiles(gemm_o), blas::Op::ConjTrans, blas::Op::NoTrans, T(1),
+                        a.read(ij), w.read(ij), T(1), x(transposed(ij)));
+        }
+      }
+    }
+  }
+}
+
+template <class T>
+void gemmComputeW2(MatrixT<T>& w2, ConstPanelT<Coord::Col, T>& w, ConstPanelT<Coord::Col, T>& x) {
+  using matrix::unwrapExtendTiles;
+  using tile::gemm_o;
+
+  const auto ex = getHpExecutor<Backend::MC>();
+
+  // Note:
+  // Not all ranks in the column always hold at least a tile in the panel Ai, but all ranks in
+  // the column are going to participate to the reduce. For them, it is important to set the
+  // partial result W2 to zero.
+  bool isW2initialized = false;
+
+  // GEMM W2 = W* . X
+  for (const auto& index_tile : w.iteratorLocal()) {
+    const T beta = !isW2initialized ? 0 : 1;
+
+    hpx::dataflow(ex, unwrapExtendTiles(gemm_o), blas::Op::ConjTrans, blas::Op::NoTrans, T(1),
+                  w.read(index_tile), x.read(index_tile), beta, w2(LocalTileIndex(0, 0)));
+
+    isW2initialized = true;  // with C++20 this can be moved into the for init-statement
+  }
+
+  if (!isW2initialized)
+    hpx::dataflow(ex, unwrapExtendTiles(tile::set0<T>), w2(LocalTileIndex(0, 0)));
```

Comment:
For better readability first set to 0 the tiles and then always use beta=1. (gemm also needs to set it to 0 if beta is 0.)